packages/sdk/src/feature.ts

```diff
@@ -47,6 +47,12 @@ export function getMatchedTraffic(
 
     return true;
   });
+
+  if (matchedTraffic && matchedTraffic.percentage > 0) {
```

Comment:
this is done here after the Array.find above, because we still want to maintain the behaviour that first matched rule (for the segments) always wins.

and then we check the percentage value being non-zero here.